### PR TITLE
add TS generic support for usePermissions hook

### PR DESCRIPTION
**Pull request by andrei9669**
there are probably other hooks as well that could use this modification but this is the one that bugs me the most in my project.
I would have made the default value "unknown", to make it more strict, but that would have probably broken a lot of code.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -13,3 +13,4 @@ examples/**/static
 examples/**/dist
 cypress/videos
 cypress/screenshots
+.idea
```
Comment (djhi) [inline .gitignore:16]: Please revert this

**Review by djhi — CHANGES_REQUESTED** [review]
Thanks!

File: .gitignore (hunk 1 above)
Comment (andrei9669) [reply .gitignore]: I have read in some older PRs that this should be in user gitignore, but why is that so? Shouldn't the repository be responsible for what is ignored and what isn't?

Comment (djhi) [reply .gitignore]: You can have a root `.gitignore` in your home directory for things specific to your tools.

**Review by djhi — APPROVED** [review]
Thanks!

**Review by fzaninotto — APPROVED** [review]

**Comment by fzaninotto** [comment]
Thanks!

**Pull request merged**